BapyCat.java

```diff
@@ -0,0 +1,775 @@
+    public TrimPattern trimPattern;
+    public Integer colour;
+
+    // thank fuck we have a builder class so we dont have to negotiate
+    // this dog ass constructor every time we make a new item
+    public CustomItem(
+            String name, Material model, Types type, HashMap<Stat, Integer> statsList,
+            List<ItemAbilities.Ability> abilities,
+            Rarity rarity, TrimMaterial trimMaterial, TrimPattern trimPattern, Integer colour) {
+
+        this.name = name;
+        this.model = model;
+        this.type = type;
+        this.statsList = statsList;
+        this.abilities = abilities;
+        this.rarity = rarity;
+        this.trimMaterial = trimMaterial;
+        this.trimPattern = trimPattern;
+        this.colour = colour;
+        // a given item ID is always the name but like NAME_LIKE_THIS
+        this.ID = name.toUpperCase().replace(" ", "_");
+        itemInstanceByID.put(this.ID, this);
+        allItems.add(this);
+
+    }
+
+    @NotNull
+    public static CustomItem getItemByID(String ID) {
+        CustomItem i = itemInstanceByID.get(ID);
+        if (i == null) return ItemTypes.empty;
+        return i;
+    }
+
+    @NotNull
+    public static int getStat(Stat stat, ItemStack item) {
+        String ID = getID(item);
+        Integer value = CustomItem.getItemByID(ID).statsList.get(stat);
+        // null means it wasnt added to the list so it dont exist mmkay
+        if (value == null) return 0;
+        return value;
+    }
+
+    @NotNull
+    public static String getID(ItemStack item) {
+        // we have to do this nonsense because air and null are not the same thing.
+        // mojang made a concentrated effort to undo the air==null thing they had going on
+        if (item == null || item.getType() == Material.AIR) return "EMPTY";
+        // goober syntax for pulling value at nbt tag
+        String ID = NBT.get(item, (Function<ReadableItemNBT, String>) nbt -> nbt.getString("ID"));
+        if (ID.isEmpty()) return "EMPTY";
+        return ID;
+    }
+
+    @NotNull
+    public static CustomItem customItemByItemStack(ItemStack item) {
+        return CustomItem.getItemByID(CustomItem.getID(item));
+    }
+
+    @NotNull
+    public static boolean isModded(ItemStack item) {
+        if (item == null || item.getType() == Material.AIR) return false;
+        Boolean result = NBT.get(item, (Function<ReadableItemNBT, Boolean>) nbt -> nbt.getBoolean("modded"));
+        if (result == null) return false;
+        return result;
+    }
+
+    @Nullable
+    public static UUID getOwner(ItemStack item) {
+        if (item == null || item.getType() == Material.AIR) return null;
+        String uuidString = NBT.get(item, (Function<ReadableItemNBT, String>) nbt -> nbt.getString("owner"));
+        if (uuidString.isEmpty()) return null;
+        return UUID.fromString(uuidString);
+    }
+
+    public boolean executeRightClickLogic(CustomPlayer player, String ID, ItemStack held) {
```

Comment:
Why is the generic item class responsible for differentiating between different item types in their behaviour